src/documents/migrations/1044_workflow_workflowaction_workflowtrigger_and_more.py

```diff
@@ -0,0 +1,494 @@
+# Generated by Django 4.2.7 on 2023-12-23 22:51
+
+import django.db.models.deletion
+import multiselectfield.db.fields
+from django.conf import settings
+from django.contrib.auth.management import create_permissions
+from django.contrib.auth.models import Group
+from django.contrib.auth.models import Permission
+from django.contrib.auth.models import User
+from django.db import migrations
+from django.db import models
+from django.db import transaction
+from django.db.models import Q
+
+from documents.models import Correspondent
+from documents.models import CustomField
+from documents.models import DocumentType
+from documents.models import StoragePath
+from documents.models import Tag
+from documents.models import Workflow
+from documents.models import WorkflowAction
+from documents.models import WorkflowTrigger
+from paperless_mail.models import MailRule
+
+
+def add_workflow_permissions(apps, schema_editor):
+    # create permissions without waiting for post_migrate signal
+    for app_config in apps.get_app_configs():
+        app_config.models_module = True
+        create_permissions(app_config, apps=apps, verbosity=0)
+        app_config.models_module = None
+
+    add_permission = Permission.objects.get(codename="add_document")
+    workflow_permissions = Permission.objects.filter(
+        codename__contains="workflow",
+    )
+
+    for user in User.objects.filter(Q(user_permissions=add_permission)).distinct():
+        user.user_permissions.add(*workflow_permissions)
+
+    for group in Group.objects.filter(Q(permissions=add_permission)).distinct():
+        group.permissions.add(*workflow_permissions)
+
+
+def remove_workflow_permissions(apps, schema_editor):
+    workflow_permissions = Permission.objects.filter(
+        codename__contains="workflow",
+    )
+
+    for user in User.objects.all():
+        user.user_permissions.remove(*workflow_permissions)
+
+    for group in Group.objects.all():
+        group.permissions.remove(*workflow_permissions)
+
+
+def migrate_consumption_templates(apps, schema_editor):  # pragma: no cover
```

Comment:
There's a test for this right?  Do we actually want it not covered?

Reply:
I was just working on this, ha! Totally agree, it's one of the biggest things I want to make sure is solid. I added tests =)